include/itkVkDiscreteGaussianImageFilter.hxx

```diff
@@ -0,0 +1,181 @@
+  // (e.g. RequestedRegion). The StreamingImageFilter changes the
+  // requested region as part of its normal processing.
+  auto localInput = TInputImage::New();
+  localInput->Graft(this->GetInput());
+
+  BaseBlurringFilterType * smoother = nullptr;
+
+  if (this->GetUseFFT())
+  {
+    smoother = static_cast<BaseBlurringFilterType *>(m_FFTBlurringFilter.GetPointer());
+    m_LastRunUsedFFT = true;
+  }
+  else
+  {
+    // Set spatial-specific parameters
+    m_SpatialBlurringFilter->SetInputBoundaryCondition(this->GetInputBoundaryCondition());
+
+    smoother = static_cast<BaseBlurringFilterType *>(m_SpatialBlurringFilter.GetPointer());
+    m_LastRunUsedFFT = false;
+  }
+
+  smoother->SetInput(localInput);
+  smoother->SetVariance(this->GetVariance());
+  smoother->SetMaximumError(this->GetMaximumError());
+  smoother->SetMaximumKernelWidth(this->GetMaximumKernelWidth());
+  smoother->SetFilterDimensionality(this->GetFilterDimensionality());
+  smoother->SetRealBoundaryCondition(this->GetRealBoundaryCondition());
+  smoother->SetUseImageSpacing(this->GetUseImageSpacing());
+
+  // Graft this filters output onto the mini-pipeline so the mini-pipeline
+  // has the correct region ivars and will write to this filters bulk data
+  // output.
+  smoother->GraftOutput(output);
+
+  // Run the mini-pipeline
+  smoother->Update();
+
+  // Graft the last output of the mini-pipeline onto this filters output so
+  // the final output has the correct region ivars and a handle to the final
+  // bulk data
+  this->GraftOutput(output);
+}
+
+
+/** Determine whether spatial (default) or FFT blurring gives the
+ *  best anticipated performance. */
+
+template <typename TInputImage, typename TOutputImage>
+bool
+VkDiscreteGaussianImageFilter<TInputImage, TOutputImage>::GetUseFFT() const
+{
+  return this->GetAnticipatedPerformanceMetric() > m_AnticipatedPerformanceMetricThreshold;
+}
+
+template <typename TInputImage, typename TOutputImage>
+float
+VkDiscreteGaussianImageFilter<TInputImage, TOutputImage>::GetAnticipatedPerformanceMetric() const
+{
+  // If input has not been set then default to 0
+  auto input = this->GetInput();
+  auto output = this->GetOutput();
+  if (input == nullptr || output == nullptr)
+    return 0.0f;
+
+  auto regionSize = output->GetRequestedRegion().GetSize();
+  auto kernelSize = GetKernelSize();
+  return itk::VkBlurringPerformanceMetric<InputImageType>::Compute(regionSize, kernelSize);
+}
+
+template <typename TInputImage, typename TOutputImage>
+auto
+VkDiscreteGaussianImageFilter<TInputImage, TOutputImage>::GetKernelRadius() const -> RadiusType
+{
+  if (this->GetUseImageSpacing() && this->GetInput() == nullptr)
+  {
+    // Invalid input set -> default radius 0
+    RadiusType radius;
+    radius.Fill(0);
+    return radius;
+  }
+
+  RadiusType radius;
+  for (unsigned int dim = 0; dim < ImageDimension; ++dim)
+  {
+    radius[dim] = DiscreteGaussianImageFilter<TInputImage, TOutputImage>::GetKernelRadius(dim);
```

Comment:
The right-hand side appears to be a call to a _static_ member function of `DiscreteGaussianImageFilter`.  Is it as it should be?

Reply:
Ah, maybe it's not static, but for a superclass....  In that case, should there be a `this->` somehow?

Reply:
... and maybe use `Superclass::` taking advantage of a prior `using Superclass = ...` statement.

Reply:
Updated to `this->Superclass::GetKernelRadius`

Reply:
Updated to `this->Superclass::GetKernelRadius`